src/hooks/useCivicProfile.js

```diff
@@ -10,18 +10,83 @@ import { RDF_PREDICATES } from '@constants';
 import useSession from './useSession';
 import useRdfCollection from './useRdfCollection';
 
+const prefix = 'urn:hud:hmis:owl#';
+
+const civicProfileConfig = {
+  firstName: {
+    type: 'string',
+    predicate: 'urn:hud:hmis:owl#FirstName'
+  },
+  lastName: {
+    type: 'string',
+    predicate: 'urn:hud:hmis:owl#LastName'
+  },
+  dateOfBirth: {
+    type: 'date',
+    predicate: 'urn:hud:hmis:owl#DOB'
+  },
+  gender: {
+    type: 'number',
+    predicate: 'urn:hud:hmis:owl#Gender'
+  },
+  lastPermanentCity: {
+    type: 'string',
+    predicate: `${prefix}LastPermanentCity`
+  },
+  lastPermanentState: {
+    type: 'string',
+    predicate: `${prefix}LastPermanentState`
+  },
+  lastPermanentStreet: {
+    type: 'string',
+    predicate: `${prefix}LastPermanentStreet`
+  },
+  lastPermanentZIP: {
+    type: 'string',
+    predicate: `${prefix}LastPermanentZIP`
+  },
+  monthsHomeless: {
+    type: 'number',
+    predicate: `${prefix}MonthsHomelessPast3Years`
+  },
+  timesHomeless: {
+    type: 'number',
+    predicate: `${prefix}TimesHomelessPast3Years`
+  },
+  timeToHousingLoss: {
+    type: 'number',
+    predicate: `${prefix}TimeToHousingLoss`
+  }
+};
+
+const convertDataToThing = (data, thingName, config) => {
+  let thingInProgress = buildThing(createThing({ name: thingName }));
+  Object.keys(data).forEach((key) => {
+    const definition = config[key];
+    if (!definition) return;
+    try {
```

Comment:
I don't think we need a try/catch here, there isn't any asynchronous calls in this particular function.

Reply:
I'm not sure what your comment about async means. Try...catch and async are orthogonal concepts. They don't really relate to each other.

Either way I refactored this section not to need it.

Reply:
Ah okay, all good.👍 

Thought try catch was mainly used for async await and promises, but I guess it's still good to use for any operations that throws an error object.